### PR TITLE
feat: implement initial chocolatey packaging workflow

### PR DESCRIPTION
**Pull request by mikeee**
# Description

This implements an initial packaging workflow for pushing to the Chocolatey community repo when a release is tagged.

The `CHOCOLATEY_TOKEN` is required to be set prior to merging - available in the 1Password vault.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #172 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Comment by mikeee** [comment]
@dapr/maintainers-cli @dapr/approvers-cli please could you offer some thoughts on this.

File: .github/scripts/chocolatey/dapr.nuspec [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    <metadata>
+        <id>dapr</id>
+        <version>dapr_cli_version</version>
+        <packageSourceUrl>https://github.com/dapr/cli/tree/master/.github/scripts/chocolatey</packageSourceUrl>
+        <owners>mikeee, dapr</owners>
+        <title>Dapr CLI (Install)</title>
+        <authors>The Dapr Authors</authors>
+        <projectUrl>https://dapr.io/</projectUrl>
+        <iconUrl>https://cdn.jsdelivr.net/gh/dapr/website/static/images/dapr.svg</iconUrl>
+        <copyright>Copyright 2021 The Dapr Authors</copyright>
```
Comment (shubham1172) [inline .github/scripts/chocolatey/dapr.nuspec:12]:
```suggestion
        <copyright>Copyright 2024 The Dapr Authors</copyright>
```

File: .github/scripts/chocolatey/tools/chocolateyInstall.ps1
Comment (shubham1172) [inline .github/scripts/chocolatey/tools/chocolateyInstall.ps1:1]: Should we also provide scripts to uninstall and upgrade? (I understand this is a first implementation, shooting my ideas).

File: .github/workflows/release_chocolatey.yml [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+      - name: Parse SHA256 from file and write to $SHA256SUM
+        shell: pwsh
+        run: |
+          $Env:SHA256SUM = (Get-Content dapr.msi.sha256 | foreach{$_.replace(" *dapr.msi", "")} | Set-Content -Path dapr.msi.sha256
+          echo "SHA256SUM=$env:SHA256SUM" >> $env:GITHUB_ENV
+
+        # Update script url
+      - name: Update install script url
+        shell: pwsh
+        run: Get-Content tools/chocolateyInstall.ps1 | foreach{$_.replace("dapr_cli_url", "https://github.com/dapr/cli/releases/download/v$env:DAPR_CLI_VERSION/dapr.msi")} | Set-Content -Path tools/chocolateyInstall.ps1
```
Comment (shubham1172) [inline .github/workflows/release_chocolatey.yml:43]: Can we change the tools directory to .github/choco/tools? This will be consistent with `winget`.

**Review by shubham1172 — COMMENTED** [review]
LGTM from a high-level and a cursory reading of the docs. Some comments.

File: .github/scripts/chocolatey/tools/chocolateyInstall.ps1
Comment (mikeee) [reply .github/scripts/chocolatey/tools/chocolateyInstall.ps1:1]: Chocolatey includes an autoinstaller - I've validated this works with the msi installer/uninstaller

**Comment by mikeee** [comment]
@dapr/maintainers-cli ping

**Comment by yaron2** [comment]
I'm in favor of this, let's try to get it merged this week

**Pull request closed without merging**